constructor-manager-api/src/constructor_manager_api/utils/worker.py

```diff
@@ -0,0 +1,110 @@
+"""Constructor updater api worker."""
+
+import json
+import logging
+from typing import List
+
+from qtpy.QtCore import QObject, QProcess, Signal  # type: ignore
+
+from constructor_manager_api.utils.conda import get_prefix_by_name
+
+
+logger = logging.getLogger(__name__)
+
+
+class ConstructorManagerWorker(QObject):
+    """A worker to run the constructor manager cli and process errors.
+
+    Parameters
+    ----------
+    args : list
+        Arguments to pass to the constructor manager.
+    detached : bool, optional
+        Run the process detached, by default ``False``.
+    """
+
+    _WORKERS: "List[ConstructorManagerWorker]" = []
+    finished = Signal(dict)
+
+    def __init__(self, args, detached=False):
+        super().__init__()
+        ConstructorManagerWorker._WORKERS.append(self)
+        self._detached = detached
+        self._program = self._executable()
+
+        if not self._program.is_file():
+            raise FileNotFoundError(f"Could not find {self._program}")
```

Comment:
I don't think this is necessary as `_executable()` looks like it will already raise such an exception.

Reply:
`True`, code refactor that left in there 🤦🏼 

Thanks!